pkg/secretutils/secret_manager.go

```diff
@@ -96,6 +103,12 @@ func (sm *SecretManager) claimSecret(secret *corev1.Secret, owner client.Object,
 				needsUpdate = true
 			}
 		}
+
+		if !utils.StringInList(secret.Finalizers, SecretsFinalizer) {
+			log.Info("setting secret finalizer")
+			secret.Finalizers = append(secret.Finalizers, SecretsFinalizer)
+			needsUpdate = true
+		}
```

Comment:
And also this one, to verify that a newly claimed secret has the finalizer set

Reply:
We do not test that secret has ownerRef set if it was intended to set it (assuming that I understand correctly and point is about fetching newly updated secret object and checking that it has finilazer set), so why we would test-verify finalizers and ignore other fields?

Reply:
Well, currently we have a finalizer dedicate test, checking that as part of the reconciliation loop the host has the finalizer set (see the [TestAddFinalizers](https://github.com/metal3-io/baremetal-operator/blob/2de940e5f2c0603495f039e90123ddaa0f549e60/controllers/metal3.io/baremetalhost_controller_test.go#L439) test). So probably it could be a good place to include also the checks on the secrets.

(we also have a test for the secret setup fields, including the OwnerReference [here](https://github.com/metal3-io/baremetal-operator/blob/2de940e5f2c0603495f039e90123ddaa0f549e60/controllers/metal3.io/baremetalhost_controller_test.go#L718) that could work as well, but the previous one has more focus on the finalization)